src/library-authoring/create-library/CreateLibrary.tsx

```diff
@@ -0,0 +1,152 @@
+import { FormattedMessage, useIntl } from '@edx/frontend-platform/i18n';
+import {
+  Alert,
+  Container,
+  Form,
+  StatefulButton,
+} from '@openedx/paragon';
+import axios from 'axios';
+import { Formik } from 'formik';
+import { useNavigate } from 'react-router-dom';
+import * as Yup from 'yup';
+
+import { REGEX_RULES } from '../../constants';
+import Header from '../../header';
+import FormikControl from '../../generic/FormikControl';
+import FormikErrorFeedback from '../../generic/FormikErrorFeedback';
+import { useOrganizationListData } from '../../generic/data/apiHooks';
+import SubHeader from '../../generic/sub-header/SubHeader';
+import { useCreateLibraryV2 } from './data/apiHooks';
+import messages from './messages';
+
+import type { CreateContentLibraryDto } from './data/types';
+
+const CreateLibrary = () => {
+  const intl = useIntl();
+  const navigate = useNavigate();
+
+  const [apiError, setApiError] = useState<React.ReactNode>();
+
+  const { noSpaceRule, specialCharsRule } = REGEX_RULES;
+  const validSlugIdRegex = /^[a-zA-Z\d]+(?:[\w -]*[a-zA-Z\d]+)*$/;
+
+  const {
+    mutateAsync,
+  } = useCreateLibraryV2();
+
+  const {
+    data: organizationListData,
+    isLoading: isOrganizationListLoading,
+  } = useOrganizationListData();
+
+  return (
+    <>
+      <Header isHiddenMainMenu />
+      <Container size="xl" className="p-4 mt-3">
+        <SubHeader
+          title={<FormattedMessage {...messages.createLibrary} />}
+        />
+        <Formik
+          initialValues={{
+            title: '',
+            org: '',
+            slug: '',
+          }}
+          validationSchema={
+            Yup.object().shape({
+              title: Yup.string()
+                .required(intl.formatMessage(messages.requiredFieldError)),
+              org: Yup.string()
+                .required(intl.formatMessage(messages.requiredFieldError))
+                .matches(
+                  specialCharsRule,
+                  intl.formatMessage(messages.disallowedCharsError),
+                )
+                .matches(noSpaceRule, intl.formatMessage(messages.noSpaceError)),
+              slug: Yup.string()
+                .required(intl.formatMessage(messages.requiredFieldError))
+                .matches(
+                  validSlugIdRegex,
+                  intl.formatMessage(messages.invalidSlugError),
+                ),
+            })
+          }
+          onSubmit={async (values: CreateContentLibraryDto) => {
+            setApiError(undefined);
+            try {
+              const data = await mutateAsync(values);
+              navigate(`/library/${data.id}`);
+            } catch (error: any) {
+              if (axios.isAxiosError(error)) {
+                setApiError((
+                  <>
+                    {error.message}
+                    <br />
+                    {JSON.stringify(error.response?.data)}
+                  </>
+                ));
```

Comment:
I think this is ok for now, but we're going to have to display this error to users better at some stage:

![image](https://github.com/open-craft/frontend-app-course-authoring/assets/7556571/88c0ae3b-acdc-4d36-8da4-98b039f5e3c0)